pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -100,6 +100,23 @@ var Generators = map[string][]tree.Builtin{
 			"Returns the input array as a set of rows",
 		),
 	},
+	"information_schema._pg_expandarray": {
+		makeGeneratorBuiltinWithReturnType(
+			tree.ArgTypes{{"input", types.AnyArray}},
+			func(args []tree.TypedExpr) types.T {
+				if len(args) == 0 {
```

Comment:
`len(args) == 0` is impossible since the function always accepts exactly one argument.

Reply:
Actually, it is needed or else this testcase will panic:

```sql
query error pq: information_schema\._pg_expandarray\(\): cannot determine type of empty array\. Consider annotating with the desired type, for example ARRAY\[\]:::int\[\]
SELECT information_schema._pg_expandarray(ARRAY[])
```

Reply:
1) That's a different case, you would test that with `len(args[0]) == 0`, not `len(args) == 0`. (but please don't do that)

2) that error is not a panic, it's a legitimate error and _it is expected_. `array[]` without an explicit type will give that error in any context, no tjust with your new built-in. Make the test case use `array[]:::int[]` and see what happens.

Reply:
I already added the test cases for `array[]:::int[]`.

So, to be clear, by removing the `if len(args) == 0`. 
```sql
SELECT information_schema._pg_expandarray(ARRAY[])
```

Results in a panic and not an error.  It's that if condition that returns the `tree.UnknownReturnType` error:
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.glob..func191(0x0, 0x0, 0x0, 0x527eee0, 0xc420502300)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/generator_builtins.go:107 +0x1dc
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.returnTypeToFixedType(0x4aee5e0, 0xc4201b3a20, 0x13)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/overload.go:265 +0x3b
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.Builtin.FixedReturnType(0x4be1540, 0xc420502320, 0x4aee5e0, 0x100, 0x3, 0x4a673dd, 0xd, 0x4a8dcc5, 0x36, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/builtin.go:125 +0x30
main.generateFunctions(0xc42012c000, 0x1, 0x2, 0xc4205001e0, 0x1f)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/cmd/docgen/funcs.go:192 +0x37e
main.init.0.func1(0xc4200ce240, 0xc420465960, 0x1, 0x2, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/cmd/docgen/funcs.go:52 +0x171
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200ce240, 0xc420465940, 0x2, 0x2, 0xc4200ce240, 0xc420465940)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:698 +0x46d
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200ce900, 0x4a75f49, 0x1e, 0xc4200ce900)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200ce900, 0x1, 0xc4200ce240)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/pkg/cmd/docgen/main.go:37 +0x27
```

Reply:
@nvanbenschoten can you advise what is the best course of action here.